### PR TITLE
Add missing doc link for operations/http-compression.html

### LINKED ISSUE #7109: Missing doc link for operations/http-compression.html
This is caused by #5624.

### PR DESCRIPTION
**Pull request by zhztheplayer**
The PR contains:
1. Add missing doc link for operations/http-compression.html; 
2. Fix magic numbers in test cases using JettyServerInitUtils.wrapWithDefaultGzipHandler

Will fix #7109 and a suggestion[1] in #5624 .

[1] https://github.com/apache/incubator-druid/pull/5624#discussion_r227032566

### CODE REVIEW

**Comment by zhztheplayer** [comment]
I will resolve Travis CI warnings later.

**Comment by zhztheplayer** [comment]
Thanks @gianm! Travis CI warnings are fixed.

File: web-console/package-lock.json [hunk 1]
```diff
@@ -3274,12 +3274,14 @@
         "balanced-match": {
           "version": "1.0.0",
           "bundled": true,
-          "dev": true
+          "dev": true,
```
Comment (clintropolis) [inline web-console/package-lock.json]: What's with the changes to this file, they seem unrelated?

Comment (zhztheplayer) [reply web-console/package-lock.json]: Oops, seems to be my mistake, thank you for the catch.
